libwayshot/src/lib.rs

```diff
@@ -93,383 +60,374 @@ pub struct CaptureRegion {
     pub height: i32,
 }
 
-/// Supported image encoding formats.
-#[derive(Debug, Copy, Clone, PartialEq, Eq)]
-pub enum EncodingFormat {
-    /// Jpeg / jpg encoder.
-    Jpg,
-    /// Png encoder.
-    Png,
-    /// Ppm encoder.
-    Ppm,
-    /// Qoi encoder.
-    Qoi,
+struct IntersectingOutput {
+    output: WlOutput,
+    region: CaptureRegion,
+    transform: Transform,
 }
 
-impl From<EncodingFormat> for image::ImageOutputFormat {
-    fn from(format: EncodingFormat) -> Self {
-        match format {
-            EncodingFormat::Jpg => image::ImageFormat::Jpeg.into(),
-            EncodingFormat::Png => image::ImageFormat::Png.into(),
-            EncodingFormat::Ppm => image::ImageFormat::Pnm.into(),
-            EncodingFormat::Qoi => image::ImageFormat::Qoi.into(),
-        }
-    }
+/// Struct to store wayland connection and globals list.
+/// # Example usage
+///
+/// ```
+/// let wayshot_connection = WayshotConnection::new().unwrap();
+/// let image_buffer = wayshot_connection.screenshot_all().unwrap();
+/// ```
+#[derive(Debug)]
+pub struct WayshotConnection {
+    conn: Connection,
+    globals: GlobalList,
```

Comment:
```suggestion
    pub conn: Connection,
    pub globals: GlobalList,
```

Reply:
Exposing these as public will be needed if the client wants to do other wayland-related operations as you can never recreate the display if you're already attached to it. This would be needed in situations when you're using `WayshotConnection::new`